### PR TITLE
Automatically copy XPC Services to correct location

### LINKED ISSUE #334: macOS: `XPC Service` dependency copied to resource folder rather embedded.
`xcodegen --version` => `1.10.0`

#### Problem:

macOS `XPC Service` dependency needs to be copied to `XPCServices` folder, but it copied to `Resources`. This cause issue on client app shown below:

```
Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named
com.mc.macOS.xpc.Service was invalidated." UserInfo={NSDebugDescription=The
connection to service named com.mc.macOS.xpc.Service was invalidated.}
```

#### Workaround:

Copy `XPC Service` bundle manually in `postbuildScripts` phase:

```
    postbuildScripts:
       - script: '[[ -r ~/.bashrc ]] && . ~/.bashrc;
       mkdir -p "${BUILT_PRODUCTS_DIR}/${XPCSERVICES_FOLDER_PATH}/";
       find "${BUILT_PRODUCTS_DIR}" -depth 1 -type d -name "*.xpc" | xargs -I{} cp -vR "{}" "${BUILT_PRODUCTS_DIR}/${XPCSERVICES_FOLDER_PATH}/"'
         name: "Run Script: Embed XPC Service"
         shell: /bin/bash
```

---

#### Example project:

[XPCDemo.zip](https://github.com/yonaskolb/XcodeGen/files/2161916/XPCDemo.zip)

### PR DESCRIPTION
**Pull request by brentleyjones**
Fixes #334.

### CODE REVIEW

**Comment by brentleyjones** [comment]
@vgorloff Let me know if this resolves your issue.

**Comment by vgorloff** [comment]
Thank you! Will check it today evening or on weekend.

File: Sources/ProjectSpec/TargetSource.swift [hunk 1]
```diff
@@ -40,6 +40,11 @@ public struct TargetSource: Equatable {
         case carbonResources
         
         public struct CopyFilesSettings: Equatable, Hashable {
+            public static let xpcServices = CopyFilesSettings(
```
Comment (yonaskolb) [inline Sources/ProjectSpec/TargetSource.swift]: I like this. We could add a similar static `modulemap` property.
Could we move it into `XcodeGenKit` as an extension though?

Comment (brentleyjones) [reply Sources/ProjectSpec/TargetSource.swift]: The main reason I did this one here, is that it shows up in the dropdown for a destination in Xcode, and if you these exact settings Xcode will render it as such.

The `modulemap` based one is more a convention, and that I would put into an extension in `XcodeGenKit`.

**Comment by vgorloff** [comment]
Local build from `brentleyjones:automatically-copy-xpc` fork fixes issue. Thanks!

```
  macOS-XPCClient:
    type: application
    platform: macOS
    settings:
      INFOPLIST_FILE: Components/macOS/XPCDemo/Client/Info.plist
      PRODUCT_BUNDLE_IDENTIFIER: com.mc.macOS.xpc.Client
      CODE_SIGN_ENTITLEMENTS: Components/macOS/XPCDemo/Client/App.entitlements
      CODE_SIGN_IDENTITY: Mac Developer
      CODE_SIGN_STYLE: Automatic
      DEVELOPMENT_TEAM: JFM85B4M73
    configFiles:
       Debug: ../WL/Shared/Configuration/App-macOS.xcconfig
       Release: ../WL/Shared/Configuration/App-macOS.xcconfig
    sources:
       - Components/macOS/XPCDemo/Common
       - Components/macOS/XPCDemo/Client
    # postbuildScripts:
    #    - script: '[[ -r ~/.bashrc ]] && . ~/.bashrc;
    #    mkdir -p "${BUILT_PRODUCTS_DIR}/${XPCSERVICES_FOLDER_PATH}/";
    #    find "${BUILT_PRODUCTS_DIR}" -depth 1 -type d -name "*.xpc" | xargs -I{} cp -vR "{}" "${BUILT_PRODUCTS_DIR}/${XPCSERVICES_FOLDER_PATH}/"'
    #      name: "Run Script: Embed XPC Service"
    #      shell: /bin/bash
    dependencies:
      - target: macOS-XPCService
        # embed: false
      - target: mcBase-macOS
```
s180802_1555561

**Comment by brentleyjones** [comment]
Glad to hear @vgorloff. Hopefully this can make the next release 😄.

File: Tests/Fixtures/TestProject/Project.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -26,6 +26,8 @@
 		BF_130062884703 /* Alamofire.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = FR_257516580010 /* Alamofire.framework */; };
 		BF_138356261076 /* InterfaceController.swift in Sources */ = {isa = PBXBuildFile; fileRef = FR_363921640403 /* InterfaceController.swift */; };
 		BF_144945303317 /* Alamofire.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = FR_410645050443 /* Alamofire.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		BF_156251722120 /* Alamofire.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = FR_257516580010 /* Alamofire.framework */; };
+		"BF_156251722120-1" /* Alamofire.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = FR_257516580010 /* Alamofire.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Comment (yonaskolb) [inline Tests/Fixtures/TestProject/Project.xcodeproj/project.pbxproj]: Seems a similar issue to that in #373

Comment (brentleyjones) [reply Tests/Fixtures/TestProject/Project.xcodeproj/project.pbxproj]: #373 will fix it.

**Comment by brentleyjones** [comment]
@yonaskolb Good for another look.

**Comment by brentleyjones** [comment]
@yonaskolb We have a lot of good changes in master. Think we can get a new release?

**Comment by yonaskolb** [comment]
@brentleyjones and @vgorloffdid do you know the answer to the question above? Question is should header files in an xpc target generate build files and be added to a build phase?

**Comment by vgorloff** [comment]
Seems it is about ObjC version of XPC Target. I will make a sample XPC ObjC project first and will look on it’s structure.

**Comment by brentleyjones** [comment]
Sorry, I don't know. Thanks @vgorloff for looking into this.

**Comment by vgorloff** [comment]
This is project specs I have for creating ObjC version of XPC Service:

```
name: XPCPingXcodeGen

targets:

  XPCPing:
    type: xpc-service
    platform: macOS
    settings:
      INFOPLIST_FILE: XPCPing/Info.plist
      PRODUCT_BUNDLE_IDENTIFIER: com.example.XPCPing
      CODE_SIGN_IDENTITY: Mac Developer
      CODE_SIGN_STYLE: Automatic
      DEVELOPMENT_TEAM: E27KE6VTF6
    sources:
       - XPCPing
       - Common
       
  XPSClient:
    type: application
    platform: macOS
    settings:
      INFOPLIST_FILE: XPSClient/Info.plist
      PRODUCT_BUNDLE_IDENTIFIER: com.example.XPSClient
      CODE_SIGN_ENTITLEMENTS: XPSClient/XPSClient.entitlements
      CODE_SIGN_IDENTITY: Mac Developer
      CODE_SIGN_STYLE: Automatic
      DEVELOPMENT_TEAM: E27KE6VTF6
    sources:
       - Common
       - XPSClient
    dependencies:
      - target: XPCPing
      
schemes:
   
   XPSClient:
     build:
       targets:
         XPSClient: all

```

[XPCPing.zip](https://github.com/yonaskolb/XcodeGen/files/2435401/XPCPing.zip)

**Comment by vgorloff** [comment]
There could be one or more header files shared between Service and Client.app (like file `XPCPingProtocol.h` from attached project). Such headers used only to define interface (protocol). Such headers seems not supposed to be part of "Copy Headers" build phase or similar.

**Comment by yonaskolb** [comment]
I ended up just removing build files that didn't below to any build phases

**Pull request merged**